src/main/java/io/zold/api/Transaction.java

```diff
@@ -37,4 +36,32 @@
      * @return The amount
      */
     long amount();
+
+    /**
+     * Smart transaction.
+     *
+     * @author Izbassar Tolegen (t.izbassar@gmail.com)
+     * @version $Id$
+     * @since 0.1
+     */
+    final class Smart implements Transaction {
+
+        /**
+         * Amount.
+         */
+        private final long amount;
```

Comment:
@t-izbassar according to the white paper:

>1. `id` : Transaction ID, an unsigned 16-bit integer, 4-symbols hex;

I think we can store this as `long` internally, but we should have a bounds check to ensure that only valid values for 16-bit integers are valid (i.e. 0 <= amount <= (2^16)-1).